### ISSUE TITLE
[Suggestion] OpenPAC/teams compatibility

### ISSUE THREAD

**Issue by Gamershy** [Thelnfamous1/Dungeons-Gear#254]
Currently, the mod assumes all players are allies and are playing together. While this is fine normally, I'm in the process of making a modpack, which will emphasize some level of PVP.

This is problematic because, currently, enchantments such as Melee Aura give ALL players strength, or Radiance heals ALL players that enter the cloud.

This suggestion is for either players in teams (via the vanilla team command) or players in parties created by OpenPAC to consider team members/party members to be allies, while considering all others to be enemies/neutral.

Players who are not in a team nor party would be treated as solo, assuming all other players to be hostile/neutral, meaning aggressive effects would essentially affect all players of a partyless or teamless player. Whereas players that are of the same team or party would share beneficial effects and ignore one another's hostile effects.

If OpenPAC is out of the question, at least allow for team functionality so the existence of PVP Arenas could at least be realized.

Also, I'd add in that it may be best to put all of this under a config named "PVPMode" or something of that nature, so that the mod can behave as it does now, but optionally support this team based pvp system.

**Comment by Patrigan** [comment]
Actually, this is straight up a bug. The code does, in fact, already contain a check for team members. I will need to take a look at it, to ensure everything is ok there.

OpenPAC is:
https://www.curseforge.com/minecraft/mc-mods/open-parties-and-claims
?

Because I am looking for claims mods to support with something I am planning in FactionCraft.

**Comment by Gamershy** [comment]
Yes, that is OoenPAC. I was just thinking of some compatibility with it so that players on a server can have direct control over who's supposed to be an ally or not.

**Comment by Patrigan** [comment]
The whole ally system is due for a rewrite. I will see if I can simplify it a lot.

**Comment by Gamershy** [comment]
Well, glad I brought it to attention. I'll be eagerly waiting the fix.
These changes will affect 1.19.2, right?

**Comment by Gamershy** [comment]
Has this been patched at all yet?
